### PR TITLE
[TTAHUB-568] Add Objective Drop Down to Goals

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Add drop down of objectives to the goals table.

## How to test

Select a Recipient from the Recipient search results. View the Goals & Objectives tab. Each goal row should now show an arrow down to expand and reveal the objectives.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-568

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/GoalsTable/GoalRow.js [hunk 1]
```diff
@@ -231,31 +237,78 @@ function GoalRow({
...
-        {' '}
-        Objective(s)
-      </td>
-      <td>
-        {
-          showContextMenu
+            : displayGoalTopics}
+        </td>
+        <td>
+          <a
```
Comment (thewatermethod) [inline frontend/src/components/GoalsTable/GoalRow.js:243]: I know you didn't originate this pattern, but I think today might be a nice day to stop using <a> as a <button>

File: frontend/src/components/GoalsTable/GoalRow.js [hunk 2]
```diff
@@ -203,24 +205,28 @@ function GoalRow({
...
-      </td>
-      <td className="text-wrap maxw-mobile">
-        {
-          showToolTip
+    <>
+      <tr className={`tta-smarthub--goal-row ${!objectivesExpanded ? 'tta-smarthub--goal-row-collapsed' : ''}`} key={`goal_row_${id}`}>
+        <td>
+          {getGoalStatusIcon()}
+          {displayStatus === 'Ceased/suspended' ? ['Ceased/ ', <br />, 'suspended'] : displayStatus}
+        </td>
```
Comment (thewatermethod) [inline frontend/src/components/GoalsTable/GoalRow.js:218]: Suspended or Ceased/Suspended? Also not sure about the way its wrapping around the icon, looks a little funky to me

File: frontend/src/components/GoalsTable/GoalRow.css [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+}
+.goals-table .usa-table .tta-smarthub--goal-row td:nth-child(5) {
+  min-width: 168px;
+}
+
+.goals-table .usa-table .tta-smarthub--goal-row-objectives-enabled {
+  cursor: pointer;
+}
+
+.goals-table .usa-table .tta-smarthub--goal-row-objectives-disabled {
```
Comment (thewatermethod) [inline frontend/src/components/GoalsTable/GoalRow.css:31]: What's pointer-events: none for?

**Review by thewatermethod — COMMENTED** [review]
Looks great! I think its pretty close

Couple of things

- Someone using just a keyboard or a screenreader needs a way to get in and out of the nested table without going all the way through, and I think the announcements need more context (you are viewing objective a of goal b, or something like that)

- I would double check w/Lauren & Kelly on the direction of your arrows, they are the opposite of the mockup
- The border that's on the mockup, the one derived from the goal color is missing

File: frontend/src/components/GoalsTable/GoalRow.css (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalsTable/GoalRow.css:31]: If there are no objectives.

Comment (thewatermethod) [reply frontend/src/components/GoalsTable/GoalRow.css:31]: Oh ok. We show the button if there are 0 objectives?

File: frontend/src/components/GoalsTable/GoalRow.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalsTable/GoalRow.js:218]: I will switch to just 'Suspended' as this was what was decided.

File: frontend/src/components/GoalsTable/ObjectiveRow.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    }
+    return <FontAwesomeIcon className="margin-right-1" size="1x" color="#c5c5c5" icon={faExclamationCircle} />;
+  };
+
+  return (
+    <>
+      <tr className="tta-smarthub--objective-row">
+        <td>
+          {title}
+          <button
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalsTable/ObjectiveRow.js:100]: Moved to top left corner.

File: frontend/src/components/GoalsTable/ObjectiveRow.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    arNumber,
+    arLegacyId,
+    arStatus,
+    endDate,
+    reasons,
+    status,
+  } = objective;
+
+  const viewOrEditLink = arStatus === 'approved' ? `/activity-reports/view/${arId}` : `/activity-reports/${arId}`;
+  const linkToAr = arLegacyId ? `/activity-reports/legacy/${arLegacyId}` : viewOrEditLink;
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalsTable/ObjectiveRow.js:26]: Definitely something we could add.

**Review by jasalisbury — COMMENTED** [review]
Two elements have an id of `goalsObjectives` (bottom and two elements up from bottom)
Screenshot from 2022-02-18 14-01-43
Also seeing some console errors; unique key, prop types, and an error on a `<select>` element (some of these might not be because of this PR)

**Comment by AdamAdHocTeam** [comment]
> Two elements have an id of `goalsObjectives` (bottom and two elements up from bottom) Screenshot from 2022-02-18 14-01-43 Also seeing some console errors; unique key, prop types, and an error on a `<select>` element (some of these might not be because of this PR)

@jasalisbury I have fixed the duplicate id, key on list, and props issues. Looks like the select issue is pre existing.

File: frontend/src/components/GoalsTable/GoalRow.js [hunk 6]
```diff
@@ -203,24 +213,42 @@ function GoalRow({
...
+          {' '}
+          (
+          {goalNumber}
+          )
+          {determineFlagStatus()}
+        </td>
+        <td className="text-wrap maxw-mobile">
+          {showToolTip
             ? (
               <Tooltip
```
Comment (thewatermethod) [inline frontend/src/components/GoalsTable/GoalRow.js:253]: The text needs to be read off, probably the prop names on this component are unclear

```suggestion
                      <Tooltip
                displayText={displayGoalTopics}
                screenReadDisplayText={false}
                buttonLabel={`click to visually reveal ${goalTopics.join(', ')}`}
                tooltipText={goalTopics.join(', ')}
                hideUnderline={false}
                svgLineTo={300}
              />
```

**Comment by AdamAdHocTeam** [comment]
> I'm seeing this when I expand objectives:
> 
> Screen Shot 2022-02-18 at 4 40 10 PM

This was fixed by adding back regionId to the select do you have latest.

**Comment by thewatermethod** [comment]
> This was fixed by adding back regionId to the select do you have latest.

Yeah ran git fetch, git pull - says "Already up to date"

**Comment by jasalisbury** [comment]
> I'm seeing this when I expand objectives:
> 
> Screen Shot 2022-02-18 at 4 40 10 PM

Do you have a dump loaded from prod? which recipient are you seeing it on?

**Comment by AdamAdHocTeam** [comment]
> > I'm seeing this when I expand objectives:
> > Screen Shot 2022-02-18 at 4 40 10 PM
> 
> Do you have a dump loaded from prod? which recipient are you seeing it on?

This is the culprit but I cannot reproduce with latest db: /recipient-tta-records/560/region/1/goals-objectives

**Review by jasalisbury — APPROVED** [review]
:ship:

**Pull request merged**